src/other/utils.jl

```diff
@@ -479,48 +479,38 @@ function _gather_groups_old_version(ds, cols, ::Val{T}; mapformats = false) wher
 end
 
 # ds assumes is grouped based on cols and groups are gathered togther
-function  _find_starts_of_groups(ds, cols::Vector, ::Val{T}; mapformats = true) where T
+function _find_starts_of_groups(ds, cols::Vector, ::Val{T}; mapformats = true) where T
```

Comment:
finding groups from  the last column was a good idea, why you dropped it?

Reply:
I think misunderstand your words before.
```
julia> [1, 3] <= 1:3
false
```
Maybe you mean in this case using `colsidx <= index(ds).sortedcols` is not correct? I can add it back and change this line.

Reply:
This needs more care, you may refer to `_check_for_fast_sort` in `src/sort/sortperm.jl`, or use this function for your check.

Reply:
It seems that `_find_starts_of_groups` is only used when `_check_for_ast_sort` returns 1 in `src/sort/sortperm.jl`. In this case, `colsidx` should be a subset of `sortedcols`.

Do you mean that I can change this `colsidx <= index(ds).sortedcols` to something like 
```
if length(colsidx) < length(scols)
        if colsidx == view(scols, 1:length(colsidx)) && rev == view(revs, 1:length(colsidx)) && mapformats == fmt
```
in  `_check_for_ast_sort`?

In this case, I can use `starts` and `ngroups[]` from the current `ds` to get better performance.

Reply:
something like that
Note that `rev` may not be needed in your function but checking if the sorted data set used format for sorting is important.
also note that you should be aware of the data set passed to your function, it can be either Dataset, or GroupBy (special view of a data set)

Reply:
Yes. As you told me, I will use functions in src/sort/groupby.jl to extract information so that it works for both `Dataset` and `GroupBy`.